packages/auth/src/graphql/plugin/index.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  ApolloServerPlugin,
+  GraphQLRequestContextDidEncounterErrors,
+  GraphQLRequestContextDidResolveOperation,
+  GraphQLRequestListener
+} from '@apollo/server'
+import { Logger } from 'pino'
+import { v4 as uuid } from 'uuid'
+
+import { GraphQLErrorCode } from '../errors'
+
+export class LoggingPlugin implements ApolloServerPlugin {
+  private logger
+
+  constructor(logger: Logger) {
+    this.logger = logger
+  }
+
+  async requestDidStart(): Promise<GraphQLRequestListener<never>> {
+    const requestId = uuid()
+    const logger = this.logger
+    let operation: string | null
+
+    return {
+      async didResolveOperation(
+        context: GraphQLRequestContextDidResolveOperation<never>
+      ) {
+        operation = context.operationName
+        logger.info({
+          requestId,
+          operation
+        })
+      },
+      async didEncounterErrors(
+        context: GraphQLRequestContextDidEncounterErrors<never>
+      ): Promise<void> {
+        if (context.errors) {
+          context.errors.forEach((error) => {
+            if (
+              error.extensions.code === GraphQLErrorCode.InternalServerError
+            ) {
+              logger.error({
+                requestId,
+                variables: context.request.variables,
+                error
+              })
```

Comment:
Do `error.extensions.code` actually get exposed in here, or is it only exposed to the client? I can't seem to trigger this error log by throwing an early error in one of our resolvers

Reply:
@mkurapov  are you testing with a `GraphQLError` like this? Im seeing this when doing this in the `getGrants` resolver.

```ts
 throw new GraphQLError('Something went wrong', {
   extensions: {
     code: GraphQLErrorCode.InternalServerError
   }
 })
 ```

Reply:
That's right @BlairCurrey, was throwing a normal error. What if we add a condition to check whether `error.extensions` exist in the first place? If that happens, we could probably assume it was an unhandled "normal" error.